etcd/v1alpha1/types_pacemakercluster.go

```diff
@@ -0,0 +1,489 @@
+	// +optional
+	ActiveStatus ResourceActiveStatusType `json:"activeStatus,omitempty"`
+
+	// node is the node where the resource is running
+	// When present, it must be a valid string between 1 and 256 characters long.
+	// When not present, the node name is unknown. This likely indicates that there is an error parsing the raw XML output.
+	// +kubebuilder:validation:MinLength=1
+	// +kubebuilder:validation:MaxLength=256
+	// +optional
+	Node string `json:"node,omitempty"`
+}
+
+// PacemakerNodeHistoryEntry represents a single operation history entry from node_history
+type PacemakerNodeHistoryEntry struct {
+	// node is the node where the operation occurred
+	// When present, it must be a valid string between 1 and 256 characters long.
+	// When not present, the resource name is unknown. This likely indicates that there is an error parsing the raw XML output.
+	// +kubebuilder:validation:MinLength=1
+	// +kubebuilder:validation:MaxLength=256
+	// +optional
+	Node string `json:"node,omitempty"`
+
+	// resource is the resource that was operated on
+	// When present, it must be a valid string between 1 and 256 characters long.
+	// When not present, the resource name is unknown. This likely indicates that there is an error parsing the raw XML output.
+	// +kubebuilder:validation:MinLength=1
+	// +kubebuilder:validation:MaxLength=256
+	// +optional
+	Resource string `json:"resource,omitempty"`
+
+	// operation is the operation that was performed (e.g., "monitor", "start", "stop")
+	// Unlike other fields, this is not an enum because while "monitor", "start" and "stop"
+	// are the most common, resource agents can define their own operations.
+	// When present, it must be a valid string between 1 and 16 characters long.
+	// When not present, the operation is unknown. This likely indicates that there is an error parsing the raw XML output.
+	// +kubebuilder:validation:MinLength=1
+	// +kubebuilder:validation:MaxLength=16
+	// +optional
+	Operation string `json:"operation,omitempty"`
+
+	// rc is the return code from the operation
+	// When present, it must be a valid integer between 0 and 2147483647 (max 32-bit int) inclusive.
+	// When not present, the return code is unknown. This likely indicates that there is an error parsing the raw XML output.
+	// +kubebuilder:validation:Minimum=0
+	// +kubebuilder:validation:Maximum=2147483647
+	// +optional
+	RC *int32 `json:"rc,omitempty"`
+
+	// rcText is the human-readable return code text (e.g., "ok", "error", "not running")
+	// When present, it must be a valid string between 1 and 16 characters long. This is a human-readable string and is not validated against any specific format.
+	// When not present, the return code text is unknown. This likely indicates that there is an error parsing the raw XML output.
+	// +kubebuilder:validation:MinLength=1
+	// +kubebuilder:validation:MaxLength=16
+	// +optional
+	RCText string `json:"rcText,omitempty"`
```

Comment:
The longest string in pacemaker aside from the deprecated strings is "insufficient privileges", which is longer than the 16 character limit here.